.buildkite/auditbeat/auditbeat-pipeline.yml

```diff
@@ -13,56 +14,146 @@ env:
-          - label: "False"
-            value: "false"
-        default: "false"
-      - select: "auditbeat - run_auditbeat_macos_tests"
-        key: "run_auditbeat_macos_tests"
-        options:
-          - label: "True"
-            value: "true"
-          - label: "False"
-            value: "false"
-        default: "false"
-      - select: "auditbeat - run_auditbeat_arm_tests"
-        key: "run_auditbeat_arm_tests"
-        options:
-          - label: "True"
-            value: "true"
-          - label: "False"
-            value: "false"
-        default: "false"
-      - select: "auditbeat - run_auditbeat_win_tests"
-        key: "run_auditbeat_win_tests"
-        options:
-          - label: "True"
-            value: "true"
-          - label: "False"
-            value: "false"
-        default: "false"
-    if: "build.source == 'ui'"
+  - group: "Auditbeat Mandatory Testing"
+    key: "auditbeat-mandatory-tests"
+
+    steps:
+      - label: ":ubuntu: Auditbeat Unit Tests"
+        command: ".buildkite/auditbeat/scripts/unit-tests.sh"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_DEFAULT_MACHINE_TYPE}"
+        artifact_paths:
+          - "auditbeat/build/*.xml"
+          - "auditbeat/build/*.json"
+
+      - label: ":rhel: Auditbeat Unit Tests"
+        command:
+          - ".buildkite/auditbeat/scripts/unit-tests.sh"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_RHEL9}"
+          machineType: "${GCP_DEFAULT_MACHINE_TYPE}"
+        artifact_paths:
+          - "auditbeat/build/*.xml"
+          - "auditbeat/build/*.json"
+
+      - label: ":windows: Auditbeat Win-2016 Auditbeat Unit Tests"
+        command: ".buildkite/scripts/win_unit_tests.ps1"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_2016}"
+          machine_type: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 200
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "auditbeat/build/*.xml"
+          - "auditbeat/build/*.json"
+
+      - label: ":windows: Auditbeat Win-2022 Unit Tests"
+        command: ".buildkite/scripts/win_unit_tests.ps1"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_2022}"
+          machine_type: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 200
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "auditbeat/build/*.xml"
+          - "auditbeat/build/*.json"
+
+      - label: ":linux: Auditbeat Crosscompile"
+        command: ".buildkite/auditbeat/scripts/crosscompile.sh"
+        env:
+          GOX_FLAGS: "-arch amd64"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_HI_PERF_MACHINE_TYPE}"
+
+  - group: "Auditbeat ARM Tests"
+    key: "auditbeat-extended-tests-arm"
+    if: build.env("BUILDKITE_PULL_REQUEST") == "false" || build.env("GITHUB_PR_LABELS") =~ /.*arm.*/
```

Comment:
Looking at the build https://buildkite.com/elastic/beats/builds/4678#018ea4a9-2fb8-4d33-b345-ef393f55c7e7 and looking at another build with the old BK pipeline. I noticed that this is not executed for every PR.

The condition should be if PR and it has the label `*arm*`.

Reply:
This condition means: We run in on a PR only in an `arm` label present. And if it's the `main` or a release branch we also run it